src/vm/gccover.cpp

```diff
@@ -72,64 +72,17 @@ static MethodDesc* getTargetMethodDesc(PCODE target)
     return nullptr;
 }
 
-
 bool IsGcCoverageInterruptInstruction(PBYTE instrPtr)
 {
-#if defined(_TARGET_ARM64_)
-    UINT32 instrVal = *reinterpret_cast<UINT32*>(instrPtr);
-    switch (instrVal)
-    {
-    case INTERRUPT_INSTR:
-    case INTERRUPT_INSTR_CALL:
-    case INTERRUPT_INSTR_PROTECT_RET:
-        return true;
-    default:
-        return false;
-    }
-#elif defined(_TARGET_ARM_)
-    
-    size_t instrLen = GetARMInstructionLength(instrPtr);
-    if (instrLen == 2)
-    {
-        UINT16 instrVal = *reinterpret_cast<UINT16*>(instrPtr);
-        switch (instrVal)
-        {
-        case INTERRUPT_INSTR:
-        case INTERRUPT_INSTR_CALL:
-        case INTERRUPT_INSTR_PROTECT_RET:
-            return true;
-        default:
-            return false;
-        }
-    }
-    else
-    {
-        _ASSERTE(instrLen == 4);
-        UINT32 instrVal = *reinterpret_cast<UINT32*>(instrPtr);
-        switch (instrVal)
-        {
-        case INTERRUPT_INSTR_32:
-        case INTERRUPT_INSTR_CALL_32:
-        case INTERRUPT_INSTR_PROTECT_RET_32:
-            return true;
-        default:
-            return false;
-        }
-    }
+    UINT32 instrVal;
+
+#ifdef _TARGET_ARMARCH_
+    instrVal = *reinterpret_cast<UINT32*>(instrPtr);
```

Comment:
This is incorrect. You are always dereferencing 32 bits, even if it's only a 16 bit instruction.

Reply:
It is fine to do so.  The first 16 bit instruction is then decode proper;y and the second 16 bit instruction is ignored.

Reply:
Are you guaranteed to not cause a page fault?

I don't like it.

At a minimum, it requires a comment.

Reply:
It would be  impossible to cause a page fault.   I can make a change to avoid doing it it that is what you prefer.

Reply:
I find it ugly, but whatever